vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/Main.java

```diff
@@ -2123,6 +2124,13 @@ private void load(HttpServerExchange exchange, UnloadedData unloadedData) {
     }
   }
 
+  public static String hashFromAnalysisId(String id) {
```

Comment:
Looking at this further, I think this method should be moved back to the DatabaseBackedProcessor class. Extracting the ID is more a function of the processor than of the server (the server handles web requests, but it's the processor that actually deals with what's in the database, ie. the hashes) and server shouldn't really be rifling around in the BaseProcessor's Patterns. 
The SingleShotProcessor also extends BaseProcessor, and if we're using it for testing as I think it was intended then it will almost certainly need access to this method.

Reply:
Should it go in BaseProcessor then? There's nothing in this method that's specific to Database-Based workflows and if SingleShotProcessor will need it then seems better to get it from a parent than a sibling

Reply:
Hmm I took a look through SingleShotProcessor and since it doesn't have access to the database, it can't do anything with a file identifier, so I was incorrect and it doesn't need access to this method.  I think it still makes sense to have it in DatabaseBackedProcessor (DBP) rather than in Main, so that DBP doesn't have to call methods in Main (e.g. DBP#446).

Reply:
Updated